### ISSUE TITLE
News: will raise error if an author is missing

### ISSUE THREAD

**Issue by davidwessman** [fsek/web#286]
Old news in the DB was missing an author and therefore generated an error in the view:
https://github.com/fsek/web/blob/master/app/views/news/_news.html.erb#L26

**Comment by jforberg** [comment]
First off, user_path is redundant, link_to will select the correct path helper automatically if passed a resource pointer.

I prefer to define a new helper called e.g. ```link``` which automatically handles nil pointers. And I also use a convention where the resource provides its own name automatically.

So my ```link``` function takes only one argument, the resource pointer and then calls e.g. ```name``` on the resource to get a display name. If passed a model class pointer it will also do the right thing.

Copy paste from one of my other apps:

```
def link obj, options = {}
    if obj.nil?
      content_tag :i, '[Invalid object]'
    elsif obj.is_a?(ActiveRecord::Base) && obj.respond_to?(:name)
      link_to obj.name, obj, options
    elsif obj.respond_to? :model_name
      link_to obj.model_name.human(:count => 2), obj
    else
      link_to obj, obj, options
    end
  end
```

Could a convention like this work for us? Checking for nil every time is very error-prone I think.

**Comment by davidwessman** [comment]
Indeed it is.

Yeah, we have added that `link_helper` as well.
I even added a similar `user_link` where I wanted it to print the name of a user instead.
Could there be a nice way to allow a favicon as well?

**Comment by jforberg** [comment]
Good. Nil-checking is the main problem of rails views so I like to do that in helpers instead, otherwise I just forget.

Icons: here is my button_link helper that I tend to use in other projects:

```ruby
  def button_link text, item, opts = {}
    args = { :class => 'block-button' }
    if opts[:class]
      args[:class] << ' ' << opts[:class]
      opts.except! :class
    end
    if opts[:icon]
      text = raw(icon_for(opts[:icon], :fixed => true)) + ' ' + text
      opts.except! :icon
    end
    args.merge! opts
    link_to text, item, args
  end
```

**Comment by davidwessman** [comment]
Closed by #287
